### PR TITLE
Enable multi-package `depend-info`, track triplets

### PR DESCRIPTION
**Pull request by dg0yt**
Enables feature control for dependencies, as in 
~~~
$ vcpkg depend-info proj[core,tiff] tiff[core]
vcpkg-cmake: 
vcpkg-cmake-config: 
liblzma: vcpkg-cmake, vcpkg-cmake-config
zlib: vcpkg-cmake
nlohmann-json: vcpkg-cmake, vcpkg-cmake-config
sqlite3[json1, tool]: vcpkg-cmake, vcpkg-cmake-config
tiff[zip, lzma]: liblzma, vcpkg-cmake, vcpkg-cmake-config, zlib
proj[tiff]: nlohmann-json, sqlite3, tiff, vcpkg-cmake, vcpkg-cmake-config
~~~

Fixes https://github.com/microsoft/vcpkg/issues/36604.

For this to work reasonably, it is necessary to make `depend-info` triplet-aware, so this is included in this PR in the following way.
- To retain current output for native targets, the default triplet is never appended.
- For non-native targets, host packages are generically labeled `:host` (instead of using the actual host triplet), to limit variation in output and to simplify recognizing host packages.
- For non-default, non-host packages, the actual triplet is append to the package name.

Examples:
~~~
$ vcpkg depend-info zlib
vcpkg-cmake: 
zlib: vcpkg-cmake
$ vcpkg depend-info zlib:arm64-linux
vcpkg-cmake: 
zlib:arm64-linux: vcpkg-cmake
$ vcpkg depend-info zlib:arm64-linux zlib --triplet x64-linux --host-triplet x64-linux-dynamic
vcpkg-cmake:host: 
zlib:arm64-linux: vcpkg-cmake:host
zlib: vcpkg-cmake:host
~~~

For valid `dot` format output, this PR depends on #1369.

### CODE REVIEW

File: src/vcpkg/commands.depend-info.cpp [hunk 1]
```diff
@@ -452,25 +457,34 @@ namespace vcpkg
...
-            msg::write_unlocalized_text(Color::success, first->package);
-            if (!features.empty())
-            {
-                msg::write_unlocalized_text(Color::warning, "[" + features + "]");
+                std::string features = Strings::join(", ", info.features);
+
+                if (strategy.show_depth)
+                {
+                    msg::write_unlocalized_text(Color::error, "(0)"); // legacy
+                }
```
Comment (dg0yt) [inline src/vcpkg/commands.depend-info.cpp:470]: Entirely pointless, but kept for compatibility: output depth at top level tree (actually digraph now) nodes.
(Suggesting removal.)

File: src/vcpkg/commands.depend-info.cpp [hunk 2]
```diff
@@ -88,59 +88,63 @@ namespace
...
-                Util::fmap(install_action.package_dependencies, [](const PackageSpec& spec) { return spec.name(); });
+                Util::fmap(install_action.package_dependencies,
+                           [&decorated_name](const PackageSpec& spec) { return decorated_name(spec); });
 
             std::unordered_set<std::string> features{install_action.feature_list.begin(),
                                                      install_action.feature_list.end()};
             features.erase(FeatureNameCore.to_string());
 
-            auto& port_name = install_action.spec.name();
+            out.push_back({decorated_name(install_action.spec), 0, std::move(features), std::move(dependencies)});
```
Comment (Thomas1664) [inline src/vcpkg/commands.depend-info.cpp:141]: Using emplace_back here would allow us to get rid of the calls to back below.

Comment (dg0yt) [reply src/vcpkg/commands.depend-info.cpp:141]: It is good enough.

**Review by BillyONeal — APPROVED** [review]
<s>This approval is subject to the documentation being updated to reflect the interface change: https://learn.microsoft.com/vcpkg/commands/depend-info
https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/commands/depend-info.md</s> done

**Review by BillyONeal — APPROVED** [review]
<s>... and an e2e test added using the new feature</s>

I thought there was already an e2e test for this, sorry.

**Comment by BillyONeal** [comment]
Thanks for the awesome new feature!

**Pull request merged**